### PR TITLE
32045 document list

### PR DESCRIPTION
**Pull request by cohnjesse**
## Description
[Original Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/32045)

On the certificate of eligibility app there is a spot on certain pages that shows the documents that the Veteran has sent us regarding their application for a certificate of eligibility. This PR adds the component to the codebase and then modifies the pages that it appears on.

## Testing done
tests will appear in a subsequent PR

## Acceptance criteria
- [x] Changes above have been made per mockups
- [x] A PR has been submitted for review

### CODE REVIEW

File: src/applications/lgy/coe/components/CoeDocumentList.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+              className="vads-u-border-bottom--1px vads-u-border-color--gray-lighter vads-u-margin-top--4 vads-u-padding-bottom--4"
+            >
+              <h3 className="vads-u-font-family--serif vads-u-margin--0">
+                {document.title}
+              </h3>
+              <p className="vads-u-margin-y--1p5">
+                Date Sent: {document.timestamp}
+              </p>
+              <a className="vads-u-margin--0" href="#">
+                <i className="fas fa-download vads-u-padding-right--1" />
```
Comment (timwright12) [inline src/applications/lgy/coe/components/CoeDocumentList.jsx:43]:
```suggestion
                <i aria-hidden="true" className="fas fa-download vads-u-padding-right--1" />
```

File: src/applications/lgy/coe/components/CoeDocumentList.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+              key={key}
+              className="vads-u-border-bottom--1px vads-u-border-color--gray-lighter vads-u-margin-top--4 vads-u-padding-bottom--4"
+            >
+              <h3 className="vads-u-font-family--serif vads-u-margin--0">
+                {document.title}
+              </h3>
+              <p className="vads-u-margin-y--1p5">
+                Date Sent: {document.timestamp}
+              </p>
+              <a className="vads-u-margin--0" href="#">
```
Comment (timwright12) [inline src/applications/lgy/coe/components/CoeDocumentList.jsx:42]: This link needs an href value

**Review by timwright12 — CHANGES_REQUESTED** [review]

File: src/applications/lgy/coe/components/CoeDocumentList.jsx (hunk 2 above)
Comment (cohnjesse) [reply src/applications/lgy/coe/components/CoeDocumentList.jsx:42]: @timwright12 We are not sure how these links are going to be generated yet so this is just a placeholder

File: src/applications/lgy/coe/components/CoeDocumentList.jsx (hunk 1 above)
Comment (cohnjesse) [reply src/applications/lgy/coe/components/CoeDocumentList.jsx]: @timwright12 Just added this

File: src/applications/lgy/coe/components/CoeDocumentList.jsx (hunk 2 above)
Comment (timwright12) [reply src/applications/lgy/coe/components/CoeDocumentList.jsx:42]: Is this releasing to prod?

File: src/applications/lgy/coe/components/CoeDocumentList.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                />
+                {document.title} {document.type}
+              </a>
+            </div>
+          );
+        })}
+      </>
+    );
+  }
+  if (props.onUploadPage) {
```
Comment (jerekshoe) [inline src/applications/lgy/coe/components/CoeDocumentList.jsx:55]: Should this be `if (!props.onUploadPage) {`? This section should show up when the status is `pending` and there are no documents to download right?

File: src/applications/lgy/coe/components/CoeDocumentList.jsx (hunk 2 above)
Comment (cohnjesse) [reply src/applications/lgy/coe/components/CoeDocumentList.jsx:42]: @timwright12 no not at this time. The back end is still being built.

File: src/applications/lgy/coe/components/CoeDocumentList.jsx (hunk 3 above)
Comment (cohnjesse) [reply src/applications/lgy/coe/components/CoeDocumentList.jsx:55]: @jerekshoe this section is only supposed to show up if the user is on the pendind-upload page.

**Review by timwright12 — APPROVED** [review]

**Review by jerekshoe — APPROVED** [review]

**Pull request merged**